security_considerations.md

```diff
@@ -10,40 +10,55 @@ if an OEM or supplier deems it necessary. In this section, we discuss several of
+
+## Checking for dependencies and conflicts
+Generally speaking, the Director repository SHOULD NOT issue a new bundle that may conflict with images known with complete certainty to have been installed on the vehicle, as listed on the last vehicle version manifest. This is because a partial bundle installation attack could mean any bundle sent after the last vehicle version manifest may have been only partly installed by ECUs. If the Director repository is not careful in handling this issue, the vehicle may end up installing conflicting images, causing ECUs to fail to interoperate.
+
+<img align="center" src="assets/images/security_1_exchange_director_vehicle.png" width="500" style="margin: 0px 20px"/>
+
+**Figure 1.** *A series of hypothetical exchanges between a director repository and a vehicle.*
+
+Consider the series of messages exchanged between a Director repository and a vehicle in Figure 1. In the first bundle of updates, the Director repository instructs ECUs A and B to install the images A-1.0.img and B-1.0.img respectively. Later, the vehicle sends a vehicle version manifest stating that these ECUs have now installed these images.
+In the second bundle of updates, the director repository instructs these ECUs to install the images A-2.0 img and B-2.0.img respectively. However, for some unknown reason, the vehicle does not send a new vehicle version manifest in response.
+In the third bundle of updates, the director repository instructs these ECUs to install the images A-3.0.img and B-3.0.img. However, the problem is that it has not received a new vehicle version manifest from the vehicle in the meantime, stating that both ECUs have installed the second bundle.
+Furthermore, the director repository knows that B-1.0 and C-3.0 conflict with each other. The only thing the director repository can be certain of is that B has installed either B-1.0 or B-2.0, and C has installed either C-1.0 or C-2.0. Thus, the director repository SHOULD NOT send the third bundle to the vehicle, because B-1.0 from the first bundle may still be installed, which would conflict with C-3.0 from the third bundle.
+Therefore, the director repository SHOULD NOT issue the third bundle, until it has received a vehicle version manifest from the vehicle that confirms that ECUs B and C have installed the second bundle, which is known to contain images that do not conflict with the third bundle.
+In conclusion, the director repository SHOULD NOT issue a new bundle until it has received confirmation, using the vehicle version manifest, that no image known to have been installed conflicts with the new images in the new bundle.
+If the director repository is not able to update a vehicle for any reason, then it SHOULD raise the issue to the OEM.
```

Comment:
Without empty lines between these lines, this all becomes one big block of text. I suggest organizing it into a few paragraphs.

Reply:
Yeah. I'll fix this. This is still a work in progress. I committed the suggestions you made so far.

Reply:
Ok. I actually broke this up into bullets. I think it works better.